File: src/main/java/cube8540/oauth/authentication/AuthenticationApplication.java
package cube8540.oauth.authentication;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.EnableAspectJAutoProxy;
import org.springframework.retry.annotation.EnableRetry;

import javax.annotation.PostConstruct;
import java.time.Clock;
import java.time.ZoneOffset;
import java.util.TimeZone;

@EnableRetry
@EnableAspectJAutoProxy
@SpringBootApplication
public class AuthenticationApplication {

    public static final ZoneOffset DEFAULT_ZONE_OFFSET = ZoneOffset.of("+09:00");

    public static final TimeZone DEFAULT_TIME_ZONE = TimeZone.getTimeZone("Asia/Seoul");

    public static final Clock DEFAULT_CLOCK = Clock.system(DEFAULT_TIME_ZONE.toZoneId());

    @PostConstruct
    void systemSetup() {
        TimeZone.setDefault(DEFAULT_TIME_ZONE);
    }

    public static void main(String[] args) {
        SpringApplication.run(AuthenticationApplication.class, args);
    }

}

File: src/main/kotlin/cube8540/oauth/authentication/AuthenticationApplication.kt
package cube8540.oauth.authentication

import java.time.Clock
import java.time.ZoneOffset
import java.util.TimeZone
import javax.annotation.PostConstruct
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.EnableAspectJAutoProxy
import org.springframework.retry.annotation.EnableRetry


@EnableRetry
@EnableAspectJAutoProxy
@SpringBootApplication
class AuthenticationApplication {

    companion object {
        @JvmField
        val DEFAULT_ZONE_OFFSET: ZoneOffset = ZoneOffset.of("+09:00")

        @JvmField
        val DEFAULT_TIME_ZONE: TimeZone = TimeZone.getTimeZone("Asia/Seoul")

        @JvmField
        val DEFAULT_CLOCK: Clock = Clock.system(DEFAULT_TIME_ZONE.toZoneId())
    }

    @PostConstruct
    fun systemSetup() {
        TimeZone.setDefault(DEFAULT_TIME_ZONE)
    }
}

fun main(args: Array<String>) {
    SpringApplication.run(AuthenticationApplication::class.java, *args)
}